src/processor/logic.py

```diff
@@ -0,0 +1,26 @@
+from itertools import chain
+
+from core import db
+
+
+def check_existing_entries(model_column, column_values):
+    """ Get a tuple of values from a model, where entries with those values
+    already exist in the database.
+
+        Args:
+            model_column (object): column to check in the form `Model.Column`
+            column_values (list): list of values to check - these
+                should ideally be unique for the model.
+
+        Returns:
+            tuple: Values of entries that are already in the database
+    """
+    return tuple(
+        chain.from_iterable(
```

Comment:
I played around with different approaches (`zip`, `map`, list comprehensions), and `chain.from_iterable` seems to do best with large datasets.